dev/service-registry/service-locator/src/test/scala/com/lightbend/lagom/registry/impl/InternalRegistrySpec.scala

```diff
@@ -0,0 +1,94 @@
+    val simpleName = "my-service"
+    val srs = new ServiceRegistryService(simpleUri)
+    val registry = new InternalRegistry(Map.empty)
+    registry.register(simpleName, srs)
+    val actual: Option[URI] = registry.lookup(simpleName, None)
+    actual should be(Some(simpleUri))
+  }
+
+  it should "add find http service when using no protoName lookup" in {
+    val httpUri = new URI("http://localhost.com:123/pathABC")
+    val httpsUri = new URI("https://localhost.com:123/pathABC")
+    val simpleName = "my-service"
+    val srs = ServiceRegistryService.of(util.Arrays.asList(httpUri, httpsUri), Collections.emptyList[ServiceAcl])
+    val registry = new InternalRegistry(Map.empty)
+    registry.register(simpleName, srs)
+    val actual: Option[URI] = registry.lookup(simpleName, Some("http"))
+    actual should be(Some(httpUri))
+  }
+
+  it should "add find http service when using `http` as protoName in lookup" in {
+    val httpUri = new URI("http://localhost.com:123/pathABC")
+    val httpsUri = new URI("https://localhost.com:123/pathABC")
+    val simpleName = "my-service"
+    val srs = ServiceRegistryService.of(util.Arrays.asList(httpUri, httpsUri), Collections.emptyList[ServiceAcl])
+    val registry = new InternalRegistry(Map.empty)
+    registry.register(simpleName, srs)
+    val actual: Option[URI] = registry.lookup(simpleName, Some("http"))
+    actual should be(Some(httpUri))
+  }
+
+  it should "add find https service when using `https` as protoName in lookup" in {
+    val httpUri = new URI("http://localhost.com:123/pathABC")
+    val httpsUri = new URI("https://localhost.com:123/pathABC")
+    val simpleName = "my-service"
+    val srs = ServiceRegistryService.of(util.Arrays.asList(httpUri, httpsUri), Collections.emptyList[ServiceAcl])
+    val registry = new InternalRegistry(Map.empty)
+    registry.register(simpleName, srs)
+    val actual: Option[URI] = registry.lookup(simpleName, Some("https"))
+    actual should be(Some(httpsUri))
+  }
+
+  it should "list all endpoints" in {
+    val httpUri = new URI("http://localhost.com:123/pathABC")
+    val httpsUri = new URI("https://localhost.com:123/pathABC")
+    val simpleName = "my-service"
+    val srs = ServiceRegistryService.of(util.Arrays.asList(httpUri, httpsUri), Collections.emptyList[ServiceAcl])
+    val registry = new InternalRegistry(Map.empty)
+    registry.register(simpleName, srs)
+    val actual: Seq[(ServiceName, Option[String], URI)] = registry.list()
+    actual.size should be(3)
+    actual.map(_._2).toSet should be(Set(None, Some("https"), Some("http")))
+  }
+
+  it should "register services with `tcp://` scheme without a portName" in {
+    val uri = new URI("tcp://localhost:4000/cas_native")
+    val registry = new InternalRegistry(Map.empty)
+    registry.register("cas_native", new ServiceRegistryService(uri))
+
+    val actual: Seq[(ServiceName, Option[String], URI)] = registry.list()
+    actual.map(_._2).toSet should be(Set(None))
+  }
+
+}
```

Comment:
Some tests asserting the `scheme`-to-`portName` logic.

This rules may need a review if the default behavior in Akka Service Discovery when doing a `Lookup("inventory, None, None)` doesn't match the rules I implemented.